extensions/BUILD.gn

```diff
@@ -58,6 +58,9 @@ source_set("extensions") {
     "renderer/xwalk_v8tools_module.cc",
     "renderer/xwalk_v8tools_module.h",
   ]
+  if (is_android) {
+    include_dirs = [ "$root_gen_dir/xwalk" ]
```

Comment:
This is a red flag. If you see yourself needing to use a hack like this there's a 99% chance you're doing something wrong, such as missing a dependency.

It's exactly what's happening here: you need this directory because you are not depending on `//xwalk/extensions/android:xwalk_core_extensions_native_jni` here even though the android files here depend on the headers generated there (the `generate_jni` template takes care of adding the right directories to the list of include dirs). In fact, it's possible that dependency is not even needed by `xwalk_runtime` itself. And it's also possible that other targets have the same problem.

The cleanest solution here is this:

```
if (is_android) {
  sources += [
    "common/android/xwalk_extension_android.cc",
    "common/android/xwalk_extension_android.h",
    "common/android/xwalk_native_extension_loader_android.cc",
    "common/android/xwalk_native_extension_loader_android.h",
  ]
  deps += [ "//xwalk/extensions/android:xwalk_core_extensions_native_jni" ]
}
```

Adding the Android files inside the `if` also brings the whole file closer to the spirit of GN's style guide, and is something already done in places like `ui/events/BUILD.gn`.